### ISSUE TITLE
Errors when building SoftHSMv2 on OSX - TestFactoryRegistry.h missing

### ISSUE THREAD

**Issue by kombatminipig** [opendnssec/SoftHSMv2#297]
I've been following the included installation guide for OSX, but always run into the following error when running ``make check``: 

```bash
Making check in src
Making check in lib
Making check in common
make[3]: Nothing to be done for `check'.
Making check in crypto
Making check in test
/Applications/Xcode.app/Contents/Developer/usr/bin/make  cryptotest
depbase=`echo cryptotest.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
	g++ -DHAVE_CONFIG_H -I. -I../../../..  -I./.. -I./../.. -I./../../common -I./../../cryptoki_compat -I./../../data_mgr -I./../../object_store -I./../../session_mgr -I./../../slot_mgr -I/usr/local/opt/openssl/include   -g -O2 -Wall -Wextra -std=c++11 -fvisibility=hidden -MT cryptotest.o -MD -MP -MF $depbase.Tpo -c -o cryptotest.o cryptotest.cpp &&\
	mv -f $depbase.Tpo $depbase.Po
cryptotest.cpp:33:10: fatal error: 'cppunit/extensions/TestFactoryRegistry.h' file not found
#include <cppunit/extensions/TestFactoryRegistry.h>
         ^
1 error generated.
make[5]: *** [cryptotest.o] Error 1
make[4]: *** [check-am] Error 2
make[3]: *** [check-recursive] Error 1
make[2]: *** [check-recursive] Error 1
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

Does anybody know what I've missed?

**Comment by mouse07410** [comment]
@LudovicRousseau I'm afraid this fix does not work. The missing file belongs to the `cppunit` package that has to be installed. In my case it's installed in `/opt/local`. Now SoftHSMv2 does not include the directories provided in the `CFLAGS`, and so fails to locate this include file:
```
. . . . .
Making check in session_mgr
Making check in test
/Applications/Xcode.app/Contents/Developer/usr/bin/make  sessionmgrtest
depbase=`echo sessionmgrtest.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
	clang++ -DHAVE_CONFIG_H -I. -I../../../..  -I./.. -I./../.. -I./../../common -I./../../crypto -I./../../data_mgr -I./../../object_store -I./../../pkcs11 -I./../../session_mgr -I./../../slot_mgr   -m64 -maes -mpclmul -mrdrnd -msse2 -mssse3 -msse4.1 -msse4.2 -std=c++11 -Os -Ofast -Wall -Wextra -fvisibility=hidden -MT sessionmgrtest.o -MD -MP -MF $depbase.Tpo -c -o sessionmgrtest.o sessionmgrtest.cpp &&\
	mv -f $depbase.Tpo $depbase.Po
sessionmgrtest.cpp:33:10: fatal error: 'cppunit/extensions/TestFactoryRegistry.h' file not
      found
#include <cppunit/extensions/TestFactoryRegistry.h>
         ^
1 error generated.
make[5]: *** [sessionmgrtest.o] Error 1
make[4]: *** [check-am] Error 2
make[3]: *** [check-recursive] Error 1
make[2]: *** [check-recursive] Error 1
make[1]: *** [check-recursive] Error 1
make: *** [check-recursive] Error 1
```

Note that everything else seems to find things located in `/opt/local` - e.g., OpenSSL, SQLite3, Botan, etc.

**Comment by LudovicRousseau** [comment]
@mouse07410 I am not sure my fix does not work.
It worked for all the Travis-CI tests (GNU/Linux only) https://travis-ci.org/opendnssec/SoftHSMv2/builds/218370709
It worked for me with cppunit installed by homebrew.

How is ``CPPUNIT_CFLAGS`` defined in your ``Makefile``?

**Comment by mouse07410** [comment]
> I am not sure my fix does not work.

I *am* sure. Unless the following can be called "working":
```
Making check in session_mgr
Making check in test
/Applications/Xcode.app/Contents/Developer/usr/bin/make  sessionmgrtest
depbase=`echo sessionmgrtest.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
	clang++ -DHAVE_CONFIG_H -I. -I../../../..  -I./.. -I./../.. -I./../../common -I./../../crypto -I./../../data_mgr -I./../../object_store -I./../../pkcs11 -I./../../session_mgr -I./../../slot_mgr   -m64 -maes -mpclmul -mrdrnd -msse2 -mssse3 -msse4.1 -msse4.2 -std=c++11 -Os -Ofast -Wall -Wextra -fvisibility=hidden -MT sessionmgrtest.o -MD -MP -MF $depbase.Tpo -c -o sessionmgrtest.o sessionmgrtest.cpp &&\
	mv -f $depbase.Tpo $depbase.Po
sessionmgrtest.cpp:33:10: fatal error: 'cppunit/extensions/TestFactoryRegistry.h' file not
      found
#include <cppunit/extensions/TestFactoryRegistry.h>
         ^
1 error generated.
make[5]: *** [sessionmgrtest.o] Error 1
make[4]: *** [check-am] Error 2
```

> It worked for me with cppunit installed by homebrew.

It does not work for me with cppunit installed by Macports, with the header files in `/opt/local/include/cppunit`, and libraries in `/opt/local/lib`. And of course `/opt/local/lib/pkgconfig/cppunit.pc`:
```
prefix=/opt/local
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: CppUnit
Description: The C++ Unit Test Library
Version: 1.13.2
Libs: -L${libdir} -lcppunit
Libs.private: -ldl
Cflags: -I${includedir}
```

> How is `CPPUNIT_CFLAGS` defined in your Makefile?
```
$ pwd
/Users/mouse/src/SoftHSMv2
$ fgrep CPPUNIT_CFLAGS Makefile
CPPUNIT_CFLAGS = -I/opt/local/include
$ 
```

P.S. Maybe you could also take a look at #303?

**Comment by LudovicRousseau** [comment]
I don't know what is special in (one of) your configuration so that cppunit needs to use ``AM_CFLAGS`` instead of ``AM_CPPFLAGS`` used for the other includes.

Regarding #303 see #302.

**Comment by mouse07410** [comment]
> I don't know what is special in (one of) your configuration so that cppunit needs to use `AM_CFLAGS` instead of `AM_CPPFLAGS` used for the other includes.

I have no idea. I will subject this machine to the "Phoenix Test" - burn all the software including OS, and re-create from scratch. If the issue persists - I'll report here.

**Comment by bellgrim** [comment]
It works for me when following the guide in https://github.com/opendnssec/SoftHSMv2/blob/develop/OSX-NOTES.md and using Homebrew. See #313

**Comment by LudovicRousseau** [comment]
My fixes have been accepted in https://github.com/opendnssec/SoftHSMv2/commit/f3783b58baf3eba1c065c9965f52149a074908e0

Maybe this issue could be fixed  by @kombatminipig or a project admin?

**Comment by mouse07410** [comment]
> It works for me when following the guide in https://github.com/opendnssec/SoftHSMv2/blob/develop/OSX-NOTES.md and using Homebrew. 

Some of us do **not** use Homebrew. I, for example, use Macports as it fits my needs better. Tying OS X correctness to having Homebrew installed does not seem a good strategy to me.

**Comment by bellgrim** [comment]
It seems that a solution may be coming in #313.

When it comes to Homebrew vs. Macports, we are not able to maintain and test multiple Mac build environments. Patches are however welcome.

**Comment by mouse07410** [comment]
See the proposed patch in #313

**Comment by bellgrim** [comment]
Fixed in #313
